north/src/npk/mod.rs

```diff
@@ -63,13 +64,8 @@ struct Signature {
 
 impl Hashes {
     #[allow(unused)]
-    pub fn from_str(file: &str) -> Result<Hashes, Error> {
-        // sha256 of signature.yaml
-        let mut hasher = sha2::Sha256::new();
-        hasher.update(file.as_bytes());
-        let _digest = format!("{:x}", hasher.finalize());
-
-        let mut docs = file.split("---");
+    pub fn from_str(file: &str, state: &State) -> Result<Hashes, Error> {
```

Comment:
Passing the complete state seems unnecessary...why not only the needed keys?

Reply:
I see...we don't know what key to pass without reading the signature file

Reply:
this breaks our tests, best run `rake check`
```
error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> north/src/npk/mod.rs:180:13
    |
66  |     pub fn from_str(file: &str, state: &State) -> Result<Hashes, Error> {
    |     ------------------------------------------------------------------- defined here
...
180 |     let s = Hashes::from_str(signature).expect("Failed to parse signature");
    |             ^^^^^^^^^^^^^^^^ --------- supplied 1 argument
    |             |
    |             expected 2 arguments
```

Reply:
Fixed - by just passing the HashMap